ignition-server/controllers/local_ignitionprovider.go

```diff
@@ -451,3 +474,57 @@ func copyFile(src, dst string) error {
 	}
 	return os.Chmod(dst, srcinfo.Mode())
 }
+
+func findArchManifest(ctx context.Context, imageRef string, pullSecret []byte, osToFind string, archToFind string) (manifestImageRef string, err error) {
+	repo, ref, err := registryclient.GetRepoSetup(ctx, imageRef, pullSecret)
+	if err != nil {
+		return "", fmt.Errorf("failed to get repo setup when getting manifest for arch: %w", err)
+	}
+
+	var srcDigest digest.Digest
+	if len(ref.ID) > 0 {
+		srcDigest = digest.Digest(ref.ID)
+	} else if len(ref.Tag) > 0 {
+		desc, err := repo.Tags(ctx).Get(ctx, ref.Tag)
```

Comment:
what's desc?

Reply:
It's a github.com/docker/distribution.Descriptor type. I think this was existing code I was reusing, but I'll rename it descriptor so it's a bit more clearer.